lib/IRGen/IRABIDetailsProvider.cpp

```diff
@@ -149,7 +149,7 @@ class IRABIDetailsProviderImpl {
     llvm::SmallVector<const ParamDecl *, 8> silParamMapping;
     for (auto param : *fd->getParameters()) {
       if (auto *tuple =
-              param->getType()->getDesugaredType()->getAs<TupleType>()) {
+              param->getInterfaceType()->getAs<TupleType>()) {
```

Comment:
Is it expected this won't handle the following? e.g

```swift
protocol P {
  associatedtype X
}
extension P where X == (Int, Int) {
  func foo(_ x: X) {}
}
```

Reply:
I guess this is a problem with anything that looks at the interface type of a parameter. I think type resolution should just map generic parameters to concrete types in this case since it's a systemic issue throughout the compiler.

However this particular check looks like some kind of hack already?

Reply:
Yeah that makes sense, just wanted to make sure the change was intentional